src/persist-client/src/read.rs

```diff
@@ -204,13 +324,71 @@ where
         timeout: Duration,
         split: SnapshotSplit,
     ) -> Result<SnapshotIter<K, V, T, D>, LocationError> {
-        todo!("{:?}{:?}", timeout, split);
+        trace!(
+            "ReadHandle::snapshot timeout={:?} split={:?}",
+            timeout,
+            split
+        );
+        // TODO: Each part should have a read capability attached to it and then
+        // we should use that here instead of the handle's capability.
+        let contents = split
+            .contents
+            .into_iter()
+            .map(|(k, v, t, d)| ((K::decode(&k), V::decode(&v)), T::decode(t), D::decode(d)))
+            .collect();
+        let iter = SnapshotIter {
+            as_of: Antichain::from(
+                split
+                    .as_of
+                    .iter()
+                    .map(|x| T::decode(*x))
+                    .collect::<Vec<_>>(),
+            ),
+            contents,
+        };
+        Ok(iter)
     }
 
     /// Returns an independent [ReadHandle] with a new [ReaderId] but the same
     /// `since`.
     pub async fn clone(&self, timeout: Duration) -> Result<Self, LocationError> {
-        todo!("{:?}", timeout);
+        trace!("ReadHandle::clone timeout={:?}", timeout);
```

Comment:
might be nicer to use `tracing::instrument` on the methods, but ill need to enable https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.SubscriberBuilder.html#method.with_span_events on our fmt subscriver

Reply:
don't bother yet, I only have these here atm so that the timeouts (irrelevant to the in-mem placeholder) don't cause unused warnings :)

Reply:
sounds good!